frontend/packages/kubevirt-plugin/src/statuses/template/template-source-status.ts

```diff
@@ -42,13 +42,40 @@ export const getTemplateSourceStatus: GetTemplateSourceStatus = ({
   pods,
   pvcs,
   dataVolumes,
+  dataSources,
 }) => {
   if (isCommonTemplate(template)) {
     const baseImageName = getParameterValue(template, TEMPLATE_BASE_IMAGE_NAME_PARAMETER);
     const baseImageNs = getParameterValue(template, TEMPLATE_BASE_IMAGE_NAMESPACE_PARAMETER);
     if (!baseImageName || !baseImageNs) {
       return null;
     }
+
+    const matchedDataSource = dataSources?.find(({ metadata }) => metadata?.name === baseImageName);
+    const matchedDataSourcePVC = pvcs?.find(
+      ({ metadata }) =>
+        metadata?.name === matchedDataSource?.spec?.source?.pvc?.name &&
+        metadata?.namespace === matchedDataSource?.spec?.source?.pvc?.namespace,
+    );
+    const matchedDataSourceDataVolume = dataVolumes?.find(
+      ({ metadata }) =>
+        metadata?.name === matchedDataSource?.spec?.source?.pvc?.name &&
+        metadata?.namespace === matchedDataSource?.spec?.source?.pvc?.namespace,
+    );
+
+    if (matchedDataSource) {
+      return {
+        source: SOURCE_TYPE.BASE_IMAGE,
+        provider: 'Red Hat',
+        isReady:
+          !matchedDataSourceDataVolume ||
```

Comment:
why is `!matchedDataSourceDataVolume` means its ready?

Reply:
removed :)